### DISCUSSION TITLE
Does next use incrementalCacheHandlerPath during build?

### DISCUSSION THREAD

**Discussion by CaptainN** [vercel/next.js#51255] [Help]
### Summary

Does next use `incrementalCacheHandlerPath` during build, or only during ISR at runtime? 

I assume it has to use it during build, so that those artifacts are there in the cache to be updated/replaced.

**Comment by better-salmon** [comment]
Hey @CaptainN! `incrementalCacheHandlerPath` is used during the build. Specifically, the module you pass to this option will be baked into the Next.js server. So, if you don’t use this option at build time, you won’t see any effect at runtime.

**Reply by CaptainN** [reply]
But does that mean it's used at build time? I have a problem in recent next.js versions where the statically generated pages have different hashes in the filenames between horizontally scaled pods, despite having the same build id specified in next.config.js. Would using this solve that for the generated static pages (not ISR - static pages, the ones generated at "generate" time (after compile)).

As a side note I would REALLY like a mode that doesn't generate ANYTHING at build time... We have ISR - let that handle everything.

**Reply by CaptainN** [reply]
(I think I'm conflating two issues here - the build chunks are the build artifacts from tracing, not the static content)

The direct question I'm asking is, if you define an `incrementalCacheHandlerPath` - does it put the *static* content, generated at build time (or with experimental-generate) in that cache, or does it still put THAT content on the disk. If you don't specify `incrementalCacheHandlerPath` the generate time content is stored on disk, and the runtime content is put in memory in the ISR cache.
